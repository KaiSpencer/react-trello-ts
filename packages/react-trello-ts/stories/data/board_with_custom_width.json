{
  "lanes": [
    {
      "id": "PLANNED",
      "title": "Planned Tasks",
      "label": "20/70",
      "style": {"width": 280},
      "cards": [
        {
          "id": "Milk",
          "title": "Buy milk",
          "label": "15 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "2 Gallons of milk at the Deli store"
        },
        {
          "id": "Plan2",
          "title": "Dispose Garbage",
          "label": "10 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Sort out recyclable and waste as needed"
        },
        {
          "id": "Plan3",
          "title": "Write Blog",
          "label": "30 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Can AI make memes?"
        },
        {
          "id": "Plan4",
          "title": "Pay Rent",
          "label": "5 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Transfer to bank account"
        }
      ]
    },
    {
      "id": "WIP",
      "title": "Work In Progress",
      "label": "10/20",
      "style": {"width": 280},
      "cards": [
        {
          "id": "Wip1",
          "title": "Clean House",
          "label": "30 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Soap wash and polish floor. Polish windows and doors. Scrap all broken glasses"
        }
      ]
    },
    {
      "id": "BLOCKED",
      "title": "Blocked",
      "label": "0/0",
      "style": {"width": 280},
      "cards": []
    },
    {
      "id": "COMPLETED",
      "title": "Completed",
      "style": {"width": 280},
      "label": "2/5",
      "cards": [
        {
          "id": "Completed1",
          "title": "Practice Meditation",
          "label": "15 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Use Headspace app"
        },
        {
          "id": "Completed2",
          "title": "Maintain Daily Journal",
          "label": "15 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Use Spreadsheet for now"
        }
      ]
    },
    {
      "id": "REPEAT",
      "title": "Repeat",
      "style": {"width": 280},
      "label": "1/1",
      "cards": [
        {
          "id": "Repeat1",
          "title": "Morning Jog",
          "label": "30 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Track using fitbit"
        }
      ]
    },
    {
      "id": "ARCHIVED",
      "title": "Archived",
      "style": {"width": 280},
      "label": "1/1",
      "cards": [
        {
          "id": "Archived1",
          "title": "Go Trekking",
          "label": "300 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Completed 10km on cycle"
        }
      ]
    },
    {
      "id": "ARCHIVED2",
      "title": "Archived2",
      "style": {"width": 280},
      "label": "1/1",
      "cards": [
        {
          "id": "Archived1",
          "title": "Go Trekking",
          "label": "300 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Completed 10km on cycle"
        }
      ]
    },
    {
      "id": "ARCHIVED3",
      "title": "Archived3",
      "style": {"width": 280},
      "label": "1/1",
      "cards": [
        {
          "id": "Archived1",
          "title": "Go Trekking",
          "label": "300 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Completed 10km on cycle"
        }
      ]
    }
  ]
}
