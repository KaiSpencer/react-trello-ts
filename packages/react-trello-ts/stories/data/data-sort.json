
{
  "lanes": [
    {
      "id": "SORTED_LANE",
      "title": "Sorted Lane",
      "label": "20/70",
      "cards": [
        {
          "id": "Card1",
          "title": "Buy milk",
          "label": "2017-12-01",
          "description": "2 Gallons of milk at the Deli store",
          "metadata": {
            "completedAt": "2017-12-01T10:00:00Z",
            "shortCode": "abc"
          }
        },
        {
          "id": "Card2",
          "title": "Dispose Garbage",
          "label": "2017-11-01",
          "description": "Sort out recyclable and waste as needed",
          "metadata": {
            "completedAt": "2017-11-01T10:00:00Z",
            "shortCode": "aaa"
          }
        },
        {
          "id": "Card3",
          "title": "Write Blog",
          "label": "2017-10-01",
          "description": "Can AI make memes?",
          "metadata": {
            "completedAt": "2017-10-01T10:00:00Z",
            "shortCode": "fa1"
          }
        },
        {
          "id": "Card4",
          "title": "Pay Rent",
          "label": "2017-09-01",
          "description": "Transfer to bank account",
          "metadata": {
            "completedAt": "2017-09-01T10:00:00Z",
            "shortCode": "ga2"
          }
        }
      ]
    }
  ]
}
