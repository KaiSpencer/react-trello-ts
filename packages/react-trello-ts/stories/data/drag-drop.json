{
  "lanes": [
    {
      "id": "PLANNED",
      "title": "Planned Tasks",
      "label": "20/70",
      "cards": [
        {
          "id": "Milk",
          "title": "Buy milk",
          "label": "15 mins",
          "description": "2 Gallons of milk at the Deli store"
        },
        {
          "id": "Plan2",
          "title": "Dispose Garbage",
          "label": "10 mins",
          "description": "Sort out recyclable and waste as needed"
        },
        {
          "id": "Plan3",
          "title": "Write Blog",
          "label": "30 mins",
          "description": "Can AI make memes?"
        },
        {
          "id": "Plan4",
          "title": "Pay Rent",
          "label": "5 mins",
          "description": "Transfer to bank account"
        }
      ]
    },
    {
      "id": "WIP",
      "title": "Work In Progress (Not Droppable)",
      "label": "10/20",
      "droppable": false,
      "cards": [
        {
          "id": "Wip1",
          "title": "Clean House",
          "label": "30 mins",
          "description": "Soap wash and polish floor. Polish windows and doors. Scrap all broken glasses"
        }
      ]
    },
    {
      "id": "COMPLETED",
      "title": "Completed (Droppable)",
      "label": "0/0",
      "style": {"width": 280},
      "cards": []
    }
  ]
}