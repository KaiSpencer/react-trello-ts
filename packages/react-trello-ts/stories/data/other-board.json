{
  "lanes": [
    {
      "id": "yesterday",
      "title": "Yesterday",
      "label": "20/70",
      "cards": [
        {
          "id": "Wip1",
          "title": "Clean House",
          "label": "30 mins",
          "description": "Soap wash and polish floor. Polish windows and doors. Scrap all broken glasses"
        }
      ]
    },
    {
      "id": "today",
      "title": "Today",
      "label": "10/20",
      "droppable": false,
      "cards": [
        {
          "id": "Milk",
          "title": "Buy milk",
          "label": "15 mins",
          "description": "2 Gallons of milk at the Deli store"
        },
        {
          "id": "Plan2",
          "title": "Dispose Garbage",
          "label": "10 mins",
          "description": "Sort out recyclable and waste as needed"
        },
        {
          "id": "Plan3",
          "title": "Write Blog",
          "label": "30 mins",
          "description": "Can AI make memes?"
        },
        {
          "id": "Plan4",
          "title": "Pay Rent",
          "label": "5 mins",
          "description": "Transfer to bank account"
        }
      ]
    },
    {
      "id": "tomorrow",
      "title": "Tomorrow",
      "label": "0/0",
      "cards": []
    }
  ]
}