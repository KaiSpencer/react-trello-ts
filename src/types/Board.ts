import {CSSProperties} from 'react'

export interface BoardData {
  lanes: Lane[]
}

export interface Lane {
  id: string
  title?: string
  label?: string
  style?: CSSProperties
  cards: Card[]
  currentPage?: number
  droppable?: boolean
  labelStyle?: CSSProperties
  cardStyle?: CSSProperties
  disallowAddingCard?: boolean
  [key: string]: any
}

export interface Card {
  id: string
  title?: string
  label?: string
  description?: string
  laneId?: string
  style?: CSSProperties
  draggable?: boolean
  [key: string]: any
}
