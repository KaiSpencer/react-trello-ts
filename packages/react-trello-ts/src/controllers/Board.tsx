import React, { createContext, FC, PropsWithChildren } from "react";
import classNames from "classnames";
import uuidv1 from "uuid/v1";
import { BoardContainer } from "./BoardContainer";
import * as DefaultComponets from "../components";
import { BoardData } from "../types/Board";
import { store } from "../store/store";

export const Board: FC<
	PropsWithChildren<{
		id?: string;
		className?: string;
		components: typeof DefaultComponets;
		data: BoardData;
		t?: any;
	}>
> = ({ data, children, className, components, id, t, ...rest }) => {
	const allClassNames = classNames("react-trello-board", className || "");

	return (
		<BoardContext.Provider value={store}>
			<components.GlobalStyle />
			<BoardContainer
				id={id || uuidv1()}
				data={data}
				components={components}
				t={t}
				className={allClassNames}
				{...rest}
			/>
		</BoardContext.Provider>
	);
};

export const BoardContext = createContext(store);
