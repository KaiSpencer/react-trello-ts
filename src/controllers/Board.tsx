import React, {createContext, FC, PropsWithChildren} from 'react'
import classNames from 'classnames'
import uuidv1 from 'uuid/v1'
import {BoardContainer} from './BoardContainer'
import * as DefaultComponets from '../components'
import {BoardData} from 'rt/types/Board'
import {store} from 'rt/store/store'

// export class _Board extends Component<{
//   id?: string
//   className?: string
//   components?: typeof DefaultComponets
//   data: BoardData
//   t?: any
// }> {
//   store: any
//   id: any
//   constructor(props) {
//     super(props)
//     this.store = this.getStore()
//     this.id = this.props.id || uuidv1()
//   }

//   getStore = () => {
//     //When you create multiple boards, unique stores are created for isolation
//     return createStore(boardReducer)
//   }

//   render() {
//     const {className, components} = this.props
//     const allClassNames = classNames('react-trello-board', className || '')
//     return (
//       <Provider store={this.store}>
//         <>
//           <components.GlobalStyle />
//           <BoardContainer id={this.id} {...this.props} className={allClassNames} />
//         </>
//       </Provider>
//     )
//   }
// }

export const Board: FC<
  PropsWithChildren<{
    id?: string
    className?: string
    components: typeof DefaultComponets
    data: BoardData
    t?: any
  }>
> = ({data, children, className, components, id, t, ...rest}) => {
  const allClassNames = classNames('react-trello-board', className || '')

  return (
    <BoardContext.Provider value={store}>
      <components.GlobalStyle />
      <BoardContainer
        id={id || uuidv1()}
        data={data}
        components={components}
        t={t}
        className={allClassNames}
        {...rest}
      />
    </BoardContext.Provider>
  )
}

export const BoardContext = createContext(store)
