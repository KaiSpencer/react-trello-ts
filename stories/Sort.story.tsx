import { Meta, StoryObj } from "@storybook/react";
import React from "react";

import Board from "../src";

import data from "./data/data-sort.json";
Board.displayName = "Board";

export default ({
	title: "Basic Functions",
	component: Board,
} satisfies Meta<typeof Board>);

type Story = StoryObj<typeof Board>;

export const SortedLane: Story = {
	name: "Sorted Lane",
	render: (args) => {
		return (
			<Board
				data={data}
				laneSortFunction={(card1, card2) =>
					new Date(card1.metadata.completedAt).getTime() -
					new Date(card2.metadata.completedAt).getTime()
				}
			/>
		);
	},
};

export const ReverseSortedLane: Story = {
	name: "Reverse Sorted Lane",
	render: (args) => {
		return (
			<Board
				data={data}
				laneSortFunction={(card1, card2) => {
					console.log("card1", card1);

					return (
						new Date(card2.metadata.completedAt).getTime() -
						new Date(card1.metadata.completedAt).getTime()
					);
				}}
			/>
		);
	},
};
