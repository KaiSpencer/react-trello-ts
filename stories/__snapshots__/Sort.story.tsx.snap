// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Functions ReverseSortedLane smoke-test 1`] = `
<div draggable="false"
     class="sc-iGgWBj iSNfA"
>
  <div>
    <div class="smooth-dnd-container horizontal">
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Sorted Lane
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              20/70
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Card1"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Buy milk
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    2017-12-01
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  2 Gallons of milk at the Deli store
                </div>
              </article>
            </div>
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Card2"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Dispose Garbage
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    2017-11-01
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  Sort out recyclable and waste as needed
                </div>
              </article>
            </div>
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Card3"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Write Blog
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    2017-10-01
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  Can AI make memes?
                </div>
              </article>
            </div>
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Card4"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Pay Rent
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    2017-09-01
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  Transfer to bank account
                </div>
              </article>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;

exports[`Basic Functions SortedLane smoke-test 1`] = `
<div draggable="false"
     class="sc-iGgWBj iSNfA"
>
  <div>
    <div class="smooth-dnd-container horizontal">
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Sorted Lane
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              20/70
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Card4"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Pay Rent
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    2017-09-01
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  Transfer to bank account
                </div>
              </article>
            </div>
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Card3"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Write Blog
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    2017-10-01
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  Can AI make memes?
                </div>
              </article>
            </div>
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Card2"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Dispose Garbage
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    2017-11-01
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  Sort out recyclable and waste as needed
                </div>
              </article>
            </div>
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Card1"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Buy milk
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    2017-12-01
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  2 Gallons of milk at the Deli store
                </div>
              </article>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
